homeassistant/components/keymitt_ble/config_flow.py

```diff
@@ -0,0 +1,152 @@
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Check if paired."""
+        errors: dict[str, str] = {}
+
+        if discovery := self._discovered_adv:
+            self._discovered_advs[discovery.address] = discovery
+        else:
+            current_addresses = self._async_current_ids()
+            for discovery_info in async_discovered_service_info(self.hass):
+                self._ble_device = discovery_info.device
+                address = discovery_info.address
+                if address in current_addresses or address in self._discovered_advs:
+                    continue
+                parsed = parse_advertisement_data(
+                    discovery_info.device, discovery_info.advertisement
+                )
+                self._discovered_adv = parsed
+                if parsed:
+                    self._discovered_advs[address] = parsed
+
+        if not self._discovered_advs:
+            return self.async_abort(reason="no_unconfigured_devices")
+
+        if user_input is not None:
+            self._name = name_from_discovery(self._discovered_adv)
+            self._bdaddr = user_input[CONF_ADDRESS]
+            await self.async_set_unique_id(self._bdaddr, raise_on_progress=False)
+            self._abort_if_unique_id_configured()
+            return await self.async_step_link()
+
+        return self.async_show_form(
+            step_id="init",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_ADDRESS): vol.In(
+                        {
+                            address: f"{parsed.data['local_name']} ({address})"
+                            for address, parsed in self._discovered_advs.items()
+                        }
+                    )
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_link(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Given a configured host, will ask the user to press the button to pair."""
+        errors: dict[str, str] = {}
+        token = randomid(32)
+        self._client = MicroBotApiClient(
+            device=self._ble_device,
+            token=token,
+        )
+        assert self._client is not None
+        if user_input is None:
+            await self._client.connect(init=True)
+            return self.async_show_form(step_id="link")
+
+        if not self._client.is_connected():
+            errors["base"] = "linking"
+        else:
+            self._client.disconnect()
+
+        if errors:
+            return self.async_show_form(step_id="link", errors=errors)
+
+        user_input[CONF_ADDRESS] = self._bdaddr
+        user_input[CONF_ACCESS_TOKEN] = token
+
+        return self.async_create_entry(title=self._name, data=user_input)  # type: ignore[arg-type]
```

Comment:
Why is the `# type: ignore[arg-type]` needed here?

Reply:
Not sure, VSC wouldn't commit without it

Reply:
ok this is failing again without the ignore. Not sure why?